resource_retriever/CMakeLists.txt

```diff
@@ -38,11 +38,51 @@ ament_export_targets(${PROJECT_NAME} HAS_LIBRARY_TARGET)
 ament_export_dependencies(ament_index_cpp)
 ament_export_dependencies(libcurl_vendor)
 
+ament_python_install_package(${PROJECT_NAME}
+  PACKAGE_DIR src/${PROJECT_NAME})
+
 if(BUILD_TESTING)
-  # TODO(wjwwood): add ament linters
+  find_package(ament_cmake_gtest REQUIRED)
+  find_package(ament_lint_auto REQUIRED)
+
+  list(APPEND AMENT_LINT_AUTO_EXCLUDE
+    ament_cmake_cppcheck
+    ament_cmake_uncrustify
+  )
+  ament_lint_auto_find_test_dependencies()
+
+  ament_cppcheck(LANGUAGE "c++")
+  ament_uncrustify(LANGUAGE "C++")
+
+  ament_add_gtest(${PROJECT_NAME}_test test/test.cpp)
+
+  if(TARGET ${PROJECT_NAME}_test)
```

Comment:
Nit, no need to check for `TARGET ${PROJECT_NAME}_test`'s existence. If `BUILD_TESTING` was true above, then the test target better exist.